modules/consensus/database.go

```diff
@@ -78,6 +78,29 @@ var (
 	SiafundPool = []byte("SiafundPool")
 )
 
+type (
+	// dbBucket represents a collection of key/value pairs inside the database.
+	dbBucket interface {
+		Get(key []byte) []byte
+	}
+
+	// dbTx represents a read-only transaction on the database that can be used
+	// for retrieving values.
+	dbTx interface {
+		Bucket(name []byte) dbBucket
+	}
+
+	// boltTxWrapper wraps a bolt.Tx so that it matches the dbBucket interface.
+	boltTxWrapper struct {
+		tx *bolt.Tx
+	}
+)
+
+// Bucket returns the dbBucket associated with the given bucket name.
+func (b boltTxWrapper) Bucket(name []byte) dbBucket {
+	return b.tx.Bucket(name)
+}
```

Comment:
And I created a separate request for database logic: #816.

I'll keep this PR open as is (rather than stripping clock/database stuff out) and then before we merge, I can collapse the commits into a single commit to clean up the change history. This PR (803) is the one that makes use of the smaller PRs to break up all the dependencies in `validHeader` and refactors it so that we can write unit tests for it.